http_server/string_contains_match_test.py

```diff
@@ -0,0 +1,55 @@
+import sys
+
+from amaranth.sim import Simulator
+from amaranth.lib.wiring import connect
+
+from string_contains_match import StringContainsMatch
+from stream_fixtures import StreamSender
+
+def test_finds_substring():
+    dut = StringContainsMatch("test")
+    sender = StreamSender(stream=dut.input)
+    sim = Simulator(dut)
+    sim.add_clock(1e-6)
+
+    async def driver(ctx):
+        ctx.set(dut.reset, 1)
+        ctx.tick()
+        ctx.set(dut.reset, 0)
+
+        while not sender.done:
+            await ctx.tick()
+        assert ctx.get(dut.accepted)
+
+    sim.add_testbench(driver)
+    sim.add_process(sender.send_passive(map(ord, "This is a test of string contains")))
+
+    with sim.write_vcd(sys.stdout):
+        sim.run_until(0.0001)

```

Comment:
Careful with `run_until` -- IIUC it's still a success if the testbench hasn't completed in that time.

Reply:
Yeah, I picked this number after looking at gtkwave with larger numbers until this works. Since I'm not doing any randomization with the StreamSender, the amount of time it'll run for will be deterministic.